src/client/Client.cc

```diff
@@ -12333,7 +12329,7 @@ void Client::_getcwd(string& dir, const UserPerm& perms)
 
     // A cwd or ancester is unlinked
     if (in->dentries.empty()) {
-      return;
+      return -ENOENT;
```

Comment:
Would this be the appropriate behaviour if a tree is deleted but the cwd is somewhere in that subtree? Maybe the result (of getcwd) if undefined, but its probably not -ENOENT.

Reply:
I checked on Linux and it will return -ENOENT:

```
$ mkdir foo
$ cd foo
~/foo$ rmdir ~/foo
~/foo$ strace -f -vv -- /bin/pwd
...
getcwd(0x628b23f84410, 4096)            = -1 ENOENT (No such file or directory)
```